### PR TITLE
Lab 2 part 3: issue with `map.removeLayer()`

### PR DESCRIPTION
**Pull request by jonoyuan**
In part3-application.js, even though `if (condition)` is not always true, `map.removeLayer()` removes all layers.

### CODE REVIEW

File: lab/lab2/js/part3-application.js [hunk 1]
```diff
@@ -28,28 +28,125 @@
...
+
+var lyr;
+var con = [];
+
 var plotData = function() {
-  /* =====================
-    Fill out this function definition
-  ===================== */
+  lyr = 0;
+
```
Comment (jonoyuan) [inline lab/lab2/js/part3-application.js]: There are 1000 layers in `map`. `condition` can be `True` or false for each iteration of `map.eachLayer()`. The counter `lyr` reflects this, reporting numbers less than 1000 at the end of `plotData()`. However, ALL map markers are removed. What's the deal with that?

File: lab/lab2/js/part3-application.js [hunk 2]
```diff
@@ -28,28 +28,125 @@
...
+      let condition = str;
+
+      if (condition) {
+        lyr = lyr + 1;
+        map.removeLayer(layer);
+      }
+    }
+  });
+
+  console.log(`Removed ${lyr} layers`);
```
Comment (jonoyuan) [inline lab/lab2/js/part3-application.js]: There are 1000 layers in `map`. `condition` can be `True` or false for each iteration of `map.eachLayer()`. The counter `lyr` reflects this, reporting numbers less than 1000 at the end of `plotData()`. However, ALL map markers are removed. What's the deal with that?

File: lab/lab2/js/part3-application.js [hunk 3]
```diff
@@ -28,28 +28,125 @@
...
+
+      let num = notbetween(opt.HOUR_OF_DA,numericField1,numericField2);
+      let str = opt.POLICE_AGC == stringField;
+
+      let condition = str;
+
+      if (condition) {
+        lyr = lyr + 1;
+        map.removeLayer(layer);
+      }
```
Comment (jonoyuan) [inline lab/lab2/js/part3-application.js:147]: There are 1000 layers in `map`. `condition` can be `true` or `false` for each iteration of `map.eachLayer()`. The counter `lyr` reflects this, reporting numbers less than 1000 at the end of `plotData()`. However, ALL map markers are removed. What's the deal with that?

File: lab/lab2/js/part3-application.js [hunk 4]
```diff
@@ -28,28 +28,125 @@
...
-    Fill out this function definition
-  ===================== */
+  lyr = 0;
+
+  map.eachLayer(function(layer) {
+    if (layer != Stamen_TonerLite) {
+      let opt = layer.options;
+
+      let num = notbetween(opt.HOUR_OF_DA,numericField1,numericField2);
+      let str = opt.POLICE_AGC == stringField;
```
Comment (moradology) [inline lab/lab2/js/part3-application.js:140]: It seems to me that you want to remove layers that *don't* match (i.e. `!==`). The other thing to watch out for is the difference between `!=` and `!==` - stringField is a string and the value stored in the layer appears to be a number

I don't think this fully resolves things, but it is a start. Maybe we can pair on this at office hours tomorrow?

Comment (moradology) [reply lab/lab2/js/part3-application.js:140]: By the way, really impressed by some of what you're doing here. Clearly going above and beyond to develop your programming vocabulary

**Pull request closed without merging**